repo/content/content_manager.go

```diff
@@ -174,7 +176,33 @@ func (bm *WriteManager) DeleteContent(ctx context.Context, contentID ID) error {
 		return err
 	}
 
-	return bm.deletePreexistingContent(ctx, bi)
+	return bm.deletePreexistingContent(ctx, bi, false)
+}
+
+// ForgetContent removes the given contentID from index. Because indices are append-only
+// and entries cannot really be removed until compaction, this changes backing blob ID
+// to PackBlobIDForgotten.
+//
+// Forget is only designed to be used for data repairs to bring back consistency in case of repository
+// corruption.
+func (bm *WriteManager) ForgetContent(ctx context.Context, contentID ID) error {
+	bm.lock()
+	defer bm.unlock()
+
+	atomic.AddInt64(&bm.revision, 1)
+
+	bm.log.Debugf("forget-content %v", contentID)
+
+	if err := bm.maybeRefreshIndexes(ctx); err != nil {
```

Comment:
Why is this refresh needed here?

Reply:
you're right, it's not, copy/paste from delete